ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java

```diff
@@ -175,6 +205,101 @@ protected void advanceToken(RemoteReplicaInfo remoteReplicaInfo, ExchangeMetadat
     remoteReplicaInfo.setLocalLagFromRemoteInBytes(exchangeMetadataResponse.localLagFromRemoteInBytes);
     // reset stored metadata response for this replica so that we send next request for metadata
     remoteReplicaInfo.setExchangeMetadataResponse(new ExchangeMetadataResponse(ServerErrorCode.No_Error));
+    logReplicationStatus(remoteReplicaInfo, exchangeMetadataResponse);
+  }
+
+  /**
+   * Write to a given file at a given offset
```

Comment:
In which case, we need random access?

Reply:
to report current status of checking the replica - curr token, curr lag.

Reply:
Can you elaborate more?
Why do we need seek and change the file position?
If we need some current token, current lag, are they supposed to be in the memory?

Reply:
And in any case you want to reset the file pointer to the start, we'd better do a "file truncate". Don't seek the position to the beginning and modify the file content. You may end up with file content garbage since the remaining content still exist.

Reply:
1/ When replicating from remote server, I want to know when the local server catches up with the remote server so that I can claim the backup checker has completed and stop the process. I don't want to guess this. I want to be sure and by printing it out to a file, I can instantly and accurately know how far its behind the remote server. And this is not just for 1 partitions but 40 or more.
2/ I already looked at truncate and it does not work as expected. It expects the file to exist. If it does not, then truncate fails. So then for the case when file is absent, i'd have check if it exists, remove the truncate option if the caller passes it and then create the file. Then the next time, i'd have to check if it exists and then honor the truncate option if the option is passed from the caller. I left a comment explaining the various combos I tried. There are several cases to handle and the code as written currently handles all cases without a lot of if-else statements. You are right about the garbage. I do see some random numbers at the end of the file but the content I require is there intact. However, the length of the text printed out each time is more or less the same such that the entire content is overwritten. We can overcome the garbage by appending plenty of spaces or newline at the end of the text.

Reply:
yes the current token and lag are in memory already, i am just printing it out

Reply:
Let me give this a bit more thought. I do like the idea of truncate because it gives a clean output. For now, I've moved it into closed-source.

Reply:
replied in closed-source